tests/Mvp.Selections.Api.Tests/Services/LicenseServiceTests.cs

```diff
@@ -0,0 +1,223 @@
+﻿using System.IO.Compression;
+using System.Net;
+using AutoFixture;
+using FluentAssertions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Mvp.Selections.Api.Model;
+using Mvp.Selections.Api.Services;
+using Mvp.Selections.Data.Repositories;
+using Mvp.Selections.Data.Repositories.Interfaces;
+using Mvp.Selections.Domain;
+using NSubstitute;
+using Xunit;
+
+namespace Mvp.Selections.Api.Tests.Services
+{
+    public class LicenseServiceTests
+    {
+        private readonly ILicenseRepository _licenseRepository;
+        private readonly LicenseService _sut;
+        private readonly ILogger<LicenseService> _logger;
+        private readonly IUserRepository _userRepository;
+
+        public LicenseServiceTests()
+        {
+            _licenseRepository = Substitute.For<ILicenseRepository>();
+            _logger = Substitute.For<ILogger<LicenseService>>();
+            _userRepository = Substitute.For<IUserRepository>();
+            _sut = new LicenseService(_licenseRepository, _userRepository, _logger);
+
+            Fixture fixture = new Fixture();
+            fixture.Behaviors
+                .OfType<ThrowingRecursionBehavior>()
+                .ToList()
+                .ForEach(b => fixture.Behaviors.Remove(b));
+            fixture.Behaviors.Add(new OmitOnRecursionBehavior());
+        }
+
+        [Fact]
+        public async Task AssignUserAsync_WithInvalidUser_ReturnBadRequest()
+        {
+            // Arrange
+            Guid licenseId = Guid.NewGuid();
+            var request = new AssignUserToLicense
+            {
+                Email = string.Empty
+            };
+
+            // Act
+            var result = await _sut.AssignLicenseToUserAsync(request, licenseId);
+
+            // Assert
+            result.StatusCode.Should().Be(HttpStatusCode.BadRequest);
+            result.Messages.Should().Contain("Email is required");
+        }
+
+        [Fact]
+        public async Task AssignUserAsync_WithValidUserAndLicense_ReturnsSuccess()
```

Comment:
Can you reorder the tests so that the Failure cases come first, and the success case last

We also need to add Test cases for the two failure exists in the method, when the user isn't found, and also when the user isn't a current MVP